odoo/tests/test_module_operations.py

```diff
@@ -148,12 +149,24 @@ def test_scripts(args):
     _logger.info("%d standalone scripts executed in %.2fs" % (len(funcs), time.time() - start_time))
 
 
+def _check_addons_path(path):
```

Comment:
You could copy the `_normalize` method of conf.py (and name it the same as it's more a build than check)

Reply:
Oh, I see, indeed. I don't even have to copy the method, I can just call it directly :+1: 
Got focused on the [config._check_addons_path](https://github.com/odoo/odoo/blob/af8b1923273ca64b330f78d8417195a9e375770e/odoo/tools/config.py#L570-L581) and didn't see that the [config._check_upgrade_path](https://github.com/odoo/odoo/blob/af8b1923273ca64b330f78d8417195a9e375770e/odoo/tools/config.py#L583-L594) actually created an util for that. No idea why both method don't use that util tho.

Reply:
wouldn't it work to give `config._check_addons_path` as a callbak when adding --addons-path argument in parse_args ?

Reply:
It' wont work didn't think about the argparse vs optparse , forget that.

Reply:
Yep that's why it is/was inspired by that code but not made it shared as stated https://github.com/odoo/odoo/pull/92995#issuecomment-1159102871

Reply:
Suggestion to make it work;

```
class CheckAddons(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        self.values = namespace
        config._check_addons_path(self, option_string, values, self)

def parse_args():
    ...
    parser.add_argument("--addons-path", "-p", type=str, action=CheckAddons,
   ...
```

Reply:
> Feel free to use this idea or not if you don't like it ;)

If it makes the code easier/better, definitely. Seems to be the case here, nice :+1: 
Suggestion applied